csv_viewer.py

```diff
@@ -118,14 +118,38 @@ def drawData(self, model, align):
         fgcolor = model.getColorAt(0, 0, 'fg')
         for row in self.visiblerows:
             for col in self.visiblecols:
-                text = model.getValueAt(row,col)
+                text = model.getValueAt(row, col)
                 self.drawText(row, col, text, fgcolor, align)
                 if bgcolor is not None:
-                    self.drawRect(row,col, color=bgcolor)
+                    self.drawRect(row, col, color=bgcolor)
 
     def drawCellEntry(self, row, col, text=None):
         pass
 
+    def adjustColumnWidths(self):
+        """Optimally adjust col widths to accomodate the longest entry
+            in each column - usually only called  on first redraw"""
+
+        try:
+            fontsize = self.thefont[1]
+        except:
+            fontsize = self.fontsize
+        scale = 8.5 * float(fontsize)/12
+        for col in range(self.cols):
+            colname = self.model.getColumnName(col)
+            if colname in self.model.columnwidths:
+                w = self.model.columnwidths[colname]
+            else:
+                w = self.cellwidth
+            # Increase maxlen by a factor of 1.5 (obtained by trial and error)
+            # to account for the wider, uppercase names in some column headers
+            maxlen = self.model.getlongestEntry(col) * 1.5
```

Comment:
Perhaps another experiment could be to change the default font to a font that has less width variation (a monospaced font like the ones we typically use on our code editors), then add the width of a couple of characters as padding, and finally, multiply `maxlen` by the width of a single character. These steps may eliminate the use of the magic number 1.5

Reply:
I'll do some more research later this week. A quick search yesterday didn't provide me with any hint as to how to do this with tk, but I'm sure with more time I can find something 😃

Reply:
OK.  If it becomes a major research project, let's track it in a new issue.